### PR TITLE
Feature/lu 3793 save validation output

### PR DESCRIPTION
**Pull request by dllmorgan**
Added story features
Removed unused classes
Refactored QL classes
Added listing.
Updated to match linting rules.

### CODE REVIEW

File: src/main/java/uk/gov/ons/collection/controller/ContributorController.java [hunk 1]
```diff
@@ -121,20 +105,27 @@ public String searchContributor(@MatrixVariable Map <String, String> searchParam
...
-    @GetMapping(value="/validationPrepConfig/{vars}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/validationPrepConfig/{vars}", produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiResponses(value = {
         @ApiResponse(code = 200, message = "Successful retrieval of all details", response = String.class)})
-    public String dataPrepConfig(@MatrixVariable Map <String, String> searchParameters){
+    public String dataPrepConfig(@MatrixVariable Map<String, String> searchParameters) {
 
         // TODO: Validate all 3 parameters have been passed through
-        log.info("API CALL!! --> /validationPrepConfig/{vars} :: " + searchParameters );
+        log.info("DEPRECATED!!!! API CALL!! --> /validationPrepConfig/{vars} :: " + searchParameters);
```
Comment (mattprice-28) [inline src/main/java/uk/gov/ons/collection/controller/ContributorController.java]: Does this still need to be here?

File: src/main/java/uk/gov/ons/collection/entity/ValidationOutputs.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            throw new InvalidJsonException("Given JSON did not contain triggered in the expected location(s): " + outputArray, err);
+        } 
+        return false;
+    }
+
+}
+
+
+
+// mutation insertOutputArray {
```
Comment (mattprice-28) [inline src/main/java/uk/gov/ons/collection/entity/ValidationOutputs.java]: Can this be removed?

File: src/main/java/uk/gov/ons/collection/service/ContributorProxy.java [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-import feign.Headers;
 import org.springframework.cloud.openfeign.FeignClient;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
 import uk.gov.ons.collection.entity.ContributorEntity;
 
-@FeignClient(name="persistence-layer")
+@FeignClient(name = "persistence-layer")
```
Comment (mattprice-28) [inline src/main/java/uk/gov/ons/collection/service/ContributorProxy.java]: Is something still using the persistence-layer?

File: src/main/java/uk/gov/ons/collection/controller/ContributorController.java (hunk 1 above)
Comment (dllmorgan) [reply src/main/java/uk/gov/ons/collection/controller/ContributorController.java]: Yes. This is called by the data prep lambda

File: src/main/java/uk/gov/ons/collection/entity/ValidationOutputs.java (hunk 2 above)
Comment (dllmorgan) [reply src/main/java/uk/gov/ons/collection/entity/ValidationOutputs.java]: Yes, but it's handy to have around in the short term as an example of a built up query.

File: src/main/java/uk/gov/ons/collection/service/ContributorProxy.java (hunk 3 above)
Comment (dllmorgan) [reply src/main/java/uk/gov/ons/collection/service/ContributorProxy.java]: Yes. Multiple services are still using the persistence layer. :(

**Pull request merged**